warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/TermFrequencyHitFunction.java

```diff
@@ -0,0 +1,689 @@
+        if (isNegated) {
+            this.hasNegatedFunctions = true;
+        }
+        
+        // Build up the list of sub queries
+        LinkedList<SubQuery> subQueries = new LinkedList<>();
+        for (String field : fields) {
+            subQueries.add(new SubQuery(subQueryId++, isNegated, field, values));
+        }
+        
+        return subQueries;
+    }
+    
+    // If a function does not contain the fields being queried, find the fields via the term frequency field-value map
+    private TreeSet<String> lookupFieldsByValues(Set<String> values) {
+        TreeSet<String> fields = new TreeSet<>();
+        for (String key : tfFVs.keySet()) {
+            for (String value : values) {
+                if (tfFVs.containsEntry(key, value)) {
+                    fields.add(key);
+                }
+            }
+        }
+        return fields;
+    }
+    
+    // A node is special if it is not the fields being searched. That is, the first node is a number or a variable 'termOffsetMap'
+    private boolean isFirstNodeSpecial(JexlNode node) {
+        if (node instanceof ASTNumberLiteral) {
+            return true;
+        } else if (node instanceof ASTReference) {
+            List<ASTIdentifier> ids = JexlASTHelper.getIdentifiers(node);
+            return ids.size() == 1 && ids.get(0).image.equals("termOffsetMap");
+        }
+        return false;
+    }
+    
+    /**
+     * Extract fields from a JexlNode. Node may be an OR node like (FIELD_A || FIELD_B).
+     *
+     * @param node
+     *            a jexl node
+     * @return a sorted set of fields
+     */
+    private TreeSet<String> parseField(JexlNode node) {
+        TreeSet<String> fields = new TreeSet<>();
+        List<ASTIdentifier> identifiers = JexlASTHelper.getIdentifiers(node);
+        for (ASTIdentifier identifier : identifiers) {
+            fields.add(identifier.image);
+        }
+        return fields;
+    }
+    
+    /**
+     * Parse the values out of a content functions arguments
+     *
+     * @param args
+     *            content function arguments as a list of Jexl nodes
+     * @param start
+     *            the start index for the values
+     * @return a list of normalized string values
+     */
+    private TreeSet<String> getValuesFromArgs(List<JexlNode> args, int start) {
+        TreeSet<String> values = new TreeSet<>();
+        for (int i = start; i < args.size(); i++) {
+            List<String> parsed = parseArg(args.get(i));
+            values.addAll(parsed);
+        }
+        return values;
+    }
+    
+    private List<String> parseArg(JexlNode node) {
+        List<String> parsed = new LinkedList<>();
+        List<Object> values = JexlASTHelper.getLiteralValues(node);
+        for (Object value : values) {
+            if (value instanceof String) {
+                parsed.add((String) value);
+            } else {
+                throw new IllegalStateException("Expected literal value to be String cast-able");
```

Comment:
How will this be handled elsewhere?

Reply:
As far as I can tell, the args in the jexl node are always strings. I have yet to find a numeric that isn't represented by a string.